plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -319,18 +322,28 @@ protected String renameTableSql(String catalogName, String remoteSchemaName, Str
     }
 
     @Override
-    protected String renameColumnSql(JdbcTableHandle handle, JdbcColumnHandle jdbcColumn, String newRemoteColumnName)
+    public void renameColumn(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle jdbcColumn, String newColumnName)
     {
         RemoteTableName remoteTableName = handle.asPlainTable().getRemoteTableName();
-        return format(
-                "sp_rename %s, %s, 'COLUMN'",
-                singleQuote(remoteTableName.getCatalogName().orElse(null), remoteTableName.getSchemaName().orElse(null), remoteTableName.getTableName(), "[" + escape(jdbcColumn.getColumnName()) + "]"),
-                "[" + newRemoteColumnName + "]");
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            String columnFrom = DOT_JOINER.join(remoteTableName.getCatalogName().orElse(null), remoteTableName.getSchemaName().orElse(null), remoteTableName.getTableName(), doubleQuote(escape(jdbcColumn.getColumnName())));
+            String newRemoteColumnName = identifierMapping.toRemoteColumnName(connection, newColumnName);
+            verifyColumnName(connection.getMetaData(), newRemoteColumnName);
+
+            try (CallableStatement renameColumn = connection.prepareCall("exec sp_rename ?, ?, 'COLUMN'")) {
+                renameColumn.setString(1, columnFrom);
+                renameColumn.setString(2, newRemoteColumnName);
+                renameColumn.execute();
+            }
+        }
+        catch (SQLException e) {
+            throw new TrinoException(JDBC_ERROR, e);
+        }
     }
 
     private static String escape(String name)
     {
-        return name.replace("'", "''");
+        return name.replace("'", "''").replace("\"", "\"\"");
```

Comment:
That's pretty non-standard to wrap in single quotes but double the double quotes.

add a comment explaining why it's so
and rename the method so it's clean in which context it can be used.

Reply:
in sql server, if we want, that double quotes present in column name, we need to double it

Reply:
just an idea: can it happen that user already provided double double quotes?

Reply:
https://github.com/trinodb/trino/pull/14272#issuecomment-1258119172